### PR TITLE
Listen on hardhat compile

### PR DESCRIPTION
**Pull request by yann300**
For testing it:

 - run a hardhat project.
 - run remixd and connect remix with the hardhat project folder.
 - run `npx run compile`.
 - check udapp in remix, the contract should be present and a label info the user that the compilation comes from hardhat
 - it's also possible to deploy these contracts.

### CODE REVIEW

File: libs/remixd/src/services/hardhatClient.ts [hunk 1]
```diff
@@ -46,4 +50,20 @@ export class HardhatClient extends PluginClient {
       })
     })
   }
+
+  listenOnHardhatCompilation () {
+    try {
+      const buildPath = utils.absolutePath('artifacts/build-info', this.currentSharedFolder)
+      const watcher = chokidar.watch(buildPath, { depth: 0, ignorePermissionErrors: true })
+      watcher.on('change', async (f: string) => {
```
Comment (ioedeveloper) [inline libs/remixd/src/services/hardhatClient.ts:58]: No error handler for change listener callback.

**Comment by bunsenstraat** [comment]
can you provide procedure for testing QA?

File: apps/remix-ide-e2e/src/helpers/hardhat_compilation.ts [hunk 2]
```diff
@@ -0,0 +1 @@
+// @ts-ignore
```
Comment (bunsenstraat) [inline apps/remix-ide-e2e/src/helpers/hardhat_compilation.ts:1]: why is this file needed?

Comment (yann300) [reply apps/remix-ide-e2e/src/helpers/hardhat_compilation.ts:1]: this is the artifact generated by hardhat, this emulate hardhat in e2e tests

**Review by bunsenstraat — APPROVED** [review]
